pkg/loop/internal/relayer/pluginprovider/chainwriter/chain_writer.go

```diff
@@ -122,8 +122,9 @@ func WithServerEncoding(version chainreader.EncodingVersion) ServerOpt {
 
 func (s *Server) SubmitTransaction(ctx context.Context, req *pb.SubmitTransactionRequest) (*emptypb.Empty, error) {
 	err := s.impl.SubmitTransaction(ctx, req.ContractName, req.Method, req.Params, req.TransactionId, req.ToAddress, TxMetaFromProto(req.Meta), req.Value.Int())
+
 	if err != nil {
-		return nil, err
+		return nil, net.WrapRPCErr(err)
```

Comment:
This is only a client-side wrapper:
```suggestion
		return nil, err
```

Reply:
sorted